### DISCUSSION TITLE
"404 There isn't a GitHub Pages site here" for blog posts generated with ABlog

### DISCUSSION THREAD

**Discussion by adriaan90** [community/community#134436] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have created a blog with Sphinx and ABlog. I have a [Github repository](https://github.com/adriaan90/adriaan90.github.io) where I add all my content. When a PR is merged, a Github Action builds the website files and copies it across to the `gh-pages` branch. The website gets updated every time the `gh-pages` branch gets updated.

When I build the website locally with `ablog build` and `ablog serve` everything works fine. When it is hosted, everything looks fine as well, except when you click on the post. Then it throws the 404 error. It also throws the 404 page when you click on `All blog posts` section at the top.

I tried looking in the HTML files that reside on the gh-pages branch, but unfortunately my HTML knowledge is too limited to see anything amiss.

I have included the `.nojekyll` file which seems to have caused most of the issues others have experienced, but I cannot find any other scenarios where some pages don't display and others do.

Any suggestions would be greatly appreciated!

Thanks!

**Comment by LiteBrite82** [comment]
@adriaan90 

`.nojekyll` needs to be in the branch that you run the build from; in this case: `main`. Put it there and run the build again.

I'm unfamiliar with ablog, does it use a 3rd party backend? (where are the blog posts stored?)

If its not using a backend, the routing issue seems like it would stem from: 
lines 63 and 74 of conf.py in the main branch. I dont see where those directories exist: "ablog/tagcloud.html", "ablog/archives.html"

**Reply by adriaan90** [reply]
Thanks for the suggestions @LiteBrite82. I added the `.nojekyll` to the `main` branch, but it seems to not have helped. Just a note - the Github Action builds the website files which will sit in a folder `_website`. These files are then copied over to `gh-pages` branch which is then published to Github Pages. So nothing from the `main` branch other than the content of `_website` is copied over for publication.

With regards to if ablog is using a 3rd party backend, I cannot say :-) Ablog creates a blog format for content that is written with reStructuredText and Sphinx. I know Sphinx is used to generate Read The Docs content. 

A working example of a blog using Ablog is https://chrisholdgraf.com/. I am taking inspiration from this blog :-) 
When you go to this blog and open a blog post, you will see the tag cloud and archives on the left side of the screen. The "ablog/tagcloud.html" and "ablog/archives.html" are extension files used by the extension `pydata_sphinx_theme` to render these functionalities.

**Comment by deep-sengupta** [comment]
URL Configuration: Check conf.py for correct URL settings.
HTML Files: Ensure post HTML files are in the right place in gh-pages.
Case Sensitivity: Match filenames and directories exactly.
.nojekyll File: Make sure it’s in the root of gh-pages.
GitHub Actions: Verify the workflow builds and deploys correctly.
Local Debugging: Build and manually check files in a temporary branch.

**Reply by adriaan90** [reply]
With regards to the actions you suggested:

**1. URL Configuration: Check conf.py for correct URL settings:** as per the ablog documentation as well as other websites that I could find, it seems to be correct
**2. HTML Files: Ensure post HTML files are in the right place in gh-pages:** from what I can see they are
**3. Case Sensitivity: Match filenames and directories exactly:** checked and it is correct
**4. .nojekyll File: Make sure it’s in the root of gh-pages:** file added in the root
**5. GitHub Actions: Verify the workflow builds and deploys correctly:** Github Action doesn't show any errors
**6. Local Debugging: Build and manually check files in a temporary branch:** created a branch `debug-gh-pages` and added all the website files in the `docs` folder. Changed the Pages setting in the repo to build the site from that folder. Error is still there

**Comment by adriaan90** [comment]
So I thought I would try a minimal example and see if I can host it. I created a branch `minimal` and pointed Github towards it for hosting the website. It seems like everything is working in terms of links, blogs etc. 

I think I am going to not change any config, add a bit of my own content (remove the generic stuff), check that everything is working fine and then merge the branch with my `main` one. 

Thanks for the help everyone!
